src/utils/common.ts

```diff
@@ -575,3 +577,28 @@ export function selectFiles(options: {
     input.click();
   });
 }
+
+export function checkPasswordStrength(password: string): ObjectValues<typeof PASSWORD_STRENGTH> {
+  if (!password) {
+    return PASSWORD_STRENGTH.weak;
+  }
+  let strength = 0;
+
+  // Length check
+  if (password.length >= 8) strength += 1;
+  if (password.length >= 12) strength += 1;
+
+  // Character variety checks
+  if (/[a-z]/.test(password)) strength += 1;
+  if (/[A-Z]/.test(password)) strength += 1;
+  if (/[0-9]/.test(password)) strength += 1;
+  if (/[^a-zA-Z0-9]/.test(password)) strength += 1;
+
+  // Determine the strength level
+  if (strength <= 3) {
+    return PASSWORD_STRENGTH.weak;
+  } if (strength <= 4) {
+    return PASSWORD_STRENGTH.medium;
```

Comment:
"1Aa." is a medium password, but it looks easy to bruteforce

Reply:
Do you allow weak passwords? I think it should be allowed.

The check logic is fine, I just suggest adjusting constants. Like, password length should have more weight than now.

Crypto users are educated, but this is an estimate we are doing for our wallet, they may assume that we have extra security layers allowing us to consider "1Aa." a medium-secure password.

Reply:
We're allowing weak passwords